test/conformance/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: MIT
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.12.1
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+enable_testing()
+
+set(UR_CONFORMANCE_TEST_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+function(add_conformance_test name)
+    set(TEST_TARGET_NAME test-${name})
+    add_executable(${TEST_TARGET_NAME}
+        ${ARGN}
+        ${UR_CONFORMANCE_TEST_DIR}/source/environment.cpp)
+    target_link_libraries(${TEST_TARGET_NAME}
+        PRIVATE
+        ${PROJECT_NAME}::loader
+        ${PROJECT_NAME}::common
+        ${PROJECT_NAME}::headers
+        GTest::gtest_main)
+    target_include_directories(${TEST_TARGET_NAME} PRIVATE
```

Comment:
You could create a `::testing` interface library for generic testing "framework" code, similar to what we have for common. This would let you avoid relative includes like this. And we will likely need some common testing utilities anyway (asserts and whatnot).